apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1787,6 +1820,82 @@ func (m *ModelManagerAPI) validateNoSeriesUpgrades(st State, force bool) error {
 	return nil
 }
 
+func (m *ModelManagerAPI) ToolVersions(entity params.Entity) (params.ToolVersionsResult, error) {
+	var err error
+	result := params.ToolVersionsResult{}
+
+	modelTag, err := names.ParseModelTag(entity.Tag)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	isModelAdmin, err := m.authorizer.HasPermission(permission.AdminAccess, modelTag)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	if !isModelAdmin && !m.isAdmin {
+		return result, apiservererrors.ErrPerm
```

Comment:
I'm not sure being an admin is essential just to be able to see what available agent binary versions there are for a model.